### PR TITLE
Speed up command declaration and resolution

### PR DESCRIPTION
**Pull request by Sata51**
- Add code style
- Refactor all file to same code style
- Update command and alias definition
- Use map to get best command resolver quickly

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+# top-most EditorConfig file
+root = true
+
+[*]
+indent_style = space
+indent_size = 2
+end_of_line = lf
+charset = utf-8
+trim_trailing_whitespace = true
+insert_final_newline = false
```
Comment (mindsers) [inline .editorconfig]: It's opinionated but I prefer having a final new line for code file

```suggestion
insert_final_newline = true
```

File: src/bot/lib/modifiers/register-commands.js [hunk 2]
```diff
@@ -1,54 +1,60 @@
...
-    return (incomingMessage, messageContext) => {
-        if (!incomingMessage.text.startsWith('!')) {
-            return { ...messageContext } // Ignore if not a command
+  const commandProcessor = new Map()
+  for (const command of commandsAndAliases) {
+    if (commandProcessor.has(command.name)) {
+      console.log(
+        `Bad command declaration. There is already a resolver for !${command.name}`,
+      )
+      throw new Error('Command redefinition')
```
Comment (mindsers) [inline src/bot/lib/modifiers/register-commands.js]: I don't think we need both the log and the error. We should merge them

File: src/bot/lib/modifiers/register-commands.js [hunk 3]
```diff
@@ -1,54 +1,60 @@
...
-                resolver: cmd.resolver
-            }
-        })
-        .filter(el => el != null)
-
-    return (incomingMessage, messageContext) => {
-        if (!incomingMessage.text.startsWith('!')) {
-            return { ...messageContext } // Ignore if not a command
+  const commandProcessor = new Map()
+  for (const command of commandsAndAliases) {
```
Comment (mindsers) [inline src/bot/lib/modifiers/register-commands.js:5]: Maybe by destructuring the `command` constant we could rename `commandProcessor` to `commands` and simplify the code a lot.

File: src/bot/lib/utils/command.js [hunk 4]
```diff
@@ -1,6 +1,7 @@
-export function command(name, resolver = () => {}) {
-    return {
-        name,
-        resolver
-    }
-}
\ No newline at end of file
+export function command(name, resolver = () => {}, alias = []) {
```
Comment (mindsers) [inline src/bot/lib/utils/command.js]: I like the `alias` idea but since it is an array we should rename it `aliases`

File: .prettierrc [hunk 5]
```diff
@@ -0,0 +1,6 @@
+{
+  "semi": false,
+  "trailingComma": "all",
+  "singleQuote": true,
+  "arrowParens": "avoid"
```
Comment (mindsers) [inline .prettierrc]: 80 characters isn't enough for our big screen I'd set it to 120 at least.

```suggestion
  "arrowParens": "avoid",
  "printWidth": 120
```

**Review by mindsers — CHANGES_REQUESTED** [review]
Thanks for taking the time to contribute to this project! It is much appreciated ❤️ 

Some small things to change and we're good! 

Open question: I'm wondering if the `command` helper could do more.

**Comment by Sata51** [comment]
I think the helper can do a little more at this time.
We can merge aliases and name. This will simplify commands registration. But some error can be lost with this method.

**Review by mindsers — APPROVED** [review]
🚀

**Pull request merged**